### ISSUE TITLE
"Random name as antagonist" doesn't un-random your name after not getting antagonist

### ISSUE THREAD

**Issue by MrMelbert** [tgstation/tgstation#67229]
Reporting client version: 514.1568

## Round ID:
183642

## Reproduction:

- Turn on "random name only when antagonist" 
- Roll antagonist
- get a random name (yay)
- round ends
- play another round
- keep the random name you got the last round you were antagonist (oh)

**Comment by Redrover1760** [comment]
Yep, this happened to me too. It also applies to random species/body/etc.

If you open up your character menu in the round, you'll see that instead of being smart about it, it literally changes the settings in the character menu and then spawns you in, instead of like, being sane, so, unsurprisingly, it does not revert.

**Comment by Ryusenshu** [comment]
i think hardcore random is saving settings too, but im not 100% sure it still does
It did completely change my 3rd Character slot and saved those settings

Feels like an oversight
